content/components/inline-message.mdx

```diff
@@ -0,0 +1,108 @@
+---
+title: Inline Message
+description: InlineMessage is used to inform the user about the result of an action within the content.
+reactId: 
+railsIds:
+figmaId: inline_message
+---
+
+import {Box} from '@primer/react'
+
+<img
+  width="960"
+  alt="Image displaying each variation of InlineMessage, from top: critical, warning, success, unavailable."
+  src="https://github.com/primer/primitives/assets/813754/f9daa010-532a-42e2-b884-45ee6263e17c"
+/>
+
+
+## Usage
+
+InlineMessages inform users about the result of an action they took. In contrast to a [Banner](./banner) an InlineMessage can be very close the where the action has taken place.
+For example below an input field, next to a button or within a table.
+
+## Anatomy
+<img
+  width="960"
+  alt="Anatomy of an InlineMessage, including the icon and the message"
+  src="https://github.com/primer/primitives/assets/813754/d854b989-ff3b-426b-9eec-0355145e534a"
+/>
+
+## Options
+
+### Variants
```

Comment:
You can still write in Markdown if you add returns before and after JSX tags, and don't indent the children of those tags. This gives you doctocat component styles for free, and it wraps headings with anchor links.

For example:

```mdx
<Box
  mb={3}
  display="flex"
  alignItems="flex-start"
  flexDirection={['column', 'column', 'column', 'column', 'row']}
  sx={{gap: 4}}
>

<img
  width="456"
  role="presentation"
  src="https://github.com/primer/design/assets/813754/c4190734-b9d2-4978-b14e-6ff0e3716efc"
/>

<div>

#### Critical

Use the `critical` variant for errors. For example, during form validation or when an action failed.

</div>

</Box>
```